### PR TITLE
re-introduce logging controlled with -v

### PR DESCRIPTION
**Pull request by DanielLeuenberger**
In the course of the blueprint refactoring, the logging functionality (logging level controlled with the -v option) got somehow lost. I find this feature useful, so I propose to re-introduce it.

### CODE REVIEW

File: tmpl/src/{{module_name}}/cli.py.j2 [hunk 1]
```diff
@@ -52,6 +55,10 @@ def print_number(ctx, *args, **kwargs) -> None:
 @click.pass_context
 def main(ctx, number: float, **kwargs) -> None:
     """Console script for test_cli_project."""
+    " initialize logging
```
Comment (ruestefa) [inline tmpl/src/{{module_name}}/cli.py.j2]: Use `#` for comments (like this, it's a syntax error).

File: tmpl/src/{{module_name}}/cli.py.j2 [hunk 2]
```diff
@@ -52,6 +55,10 @@ def print_number(ctx, *args, **kwargs) -> None:
...
+    logging.getLogger(__name__)
```
Comment (ruestefa) [inline tmpl/src/{{module_name}}/cli.py.j2:59]: Need to import `logging`.

File: tmpl/src/{{module_name}}/cli.py.j2 [hunk 3]
```diff
@@ -2,6 +2,9 @@
 # Third-party
```
Comment (ruestefa) [inline tmpl/src/{{module_name}}/cli.py.j2:2]: Need to import `logging`.

**Review by ruestefa — COMMENTED** [review]
I've flagged some errors (missing import, wrong comment symbol).

~~However, the refactoring appears to have not only broken the logging level, but the entire CLI! W/o calling the CLI in the now missing `if __name__ == "__main__":` block, the CLI is never executed! That likely snuck by the automated testing b/c calling `flying_circus.cli` doesn't throw an error, even w/ flags, but just silently succeeds in doing exactly nothing. I'll commit a fix in a few minutes.~~

Update: Sorry, I mixed this up, actually the `if __name__ == "__main__":` block is not required if a command is defined in `pyproject.toml`. However, the respective line is actually commented, so no command is defined and hence there is no way to test whether the sample application actually works...

File: tmpl/src/{{module_name}}/cli.py.j2 (hunk 3 above)
Comment (DanielLeuenberger) [reply tmpl/src/{{module_name}}/cli.py.j2:2]: Thanks, Stefan, and sorry to bother you with my errors. The review request of the PR gets out before the github actions with all the tests have finished. How can we avoid this?

Comment (DanielLeuenberger) [reply tmpl/src/{{module_name}}/cli.py.j2:2]: I will do the fixes until all tests are successful.

Comment (ruestefa) [reply tmpl/src/{{module_name}}/cli.py.j2:2]: Ah, so I was just too fast... :) I don't know whether there's a more proper way to delay notifications, but you can just skip the reviewers when opening the PR and assign them retroactively once the tests validate.

**Comment by ruestefa** [comment]
I've applied a few small fixes and introduced a callback function to set the logging level directly via click so nothing needs to be done w/i `main()`.

**Comment by ruestefa** [comment]
I've also uncommented the declaration of the command in `pyproject.toml` so the sample application can actually be run (and tested).

**Comment by ruestefa** [comment]
I realized that the application is indeed tested via the click unit test interface, so the tests are run w/ `pytest`. They even caught a bug in my new verbosity decorator function (forgot `return value`).

**Comment by DanielLeuenberger** [comment]
> Note that I changed the logger from module-specific to root (`getLogger()` instead of `getLogger(__name__)`) since that works for sure while named loggers may have unintended consequences if not done correctly (e.g., https://stackoverflow.com/a/50755200/4419816).

**Review by ruestefa — APPROVED** [review]

**Review by clairemerker — APPROVED** [review]
Fine for me.

**Pull request merged**